File: Avaliação de Checagem 1/media_mpi.c
#include <mpi.h>
#include <stdio.h>

int main(int argc, char** argv) {
    int rank, size;
    int dado;

    MPI_Init(&argc, &argv);
    MPI_Comm_rank(MPI_COMM_WORLD, &rank); 
    MPI_Comm_size(MPI_COMM_WORLD, &size); 

    if (size < 2) {
        if (rank == 0) {
            printf("Execute com pelo menos 2 processos.\n");
        }
        MPI_Finalize();
        return 0;
    }

    if (rank == 0) {
        dado = 42;
        printf("Processo %d enviando dado: %d para processo 1\n", rank, dado);
        MPI_Send(&dado, 1, MPI_INT, 1, 0, MPI_COMM_WORLD);
    } else if (rank == 1) {
        MPI_Recv(&dado, 1, MPI_INT, 0, 0, MPI_COMM_WORLD, MPI_STATUS_IGNORE);
        printf("Processo %d recebeu dado: %d do processo 0\n", rank, dado);
    }

    MPI_Finalize(); 
    return 0;
}

File: Avaliação de Checagem 1/media_mpi.py
from mpi4py import MPI

comm = MPI.COMM_WORLD
rank = comm.Get_rank()
size = comm.Get_size()

if size < 2:
    if rank == 0:
        print("Execute com pelo menos 2 processos.")
else:
    if rank == 0:
        dado = 42
        print(f"Processo {rank} enviando dado: {dado} para processo 1")
        comm.send(dado, dest=1, tag=0)
    elif rank == 1:
        dado = comm.recv(source=0, tag=0)
        print(f"Processo {rank} recebeu dado: {dado} do processo 0")
